File: nom-api/Nom.Orch/Models/Recipe/RecipeDashboardAnalyticsModel.cs
namespace Nom.Orch.Models.Recipe
{
    public class RecipeDashboardAnalyticsModel
    {
        public int TotalRecipes { get; set; }
        public Dictionary<string, int> RecipesByStatus { get; set; } = new();
        public List<RecipeSummaryModel> TopRatedRecipes { get; set; } = new();
        public List<RecipeSummaryModel> RecentlyCreated { get; set; } = new();
        public List<IngredientUsageModel> MostUsedIngredients { get; set; } = new();
    }
}

File: nom-ui/src/app/recipe/models/recipe-dashboard-analytics.model.ts
export interface RecipeSummaryModel {
    id: number;
    name: string;
    rating: number;
    createdDate: Date;
    imageUrl?: string;
}

export interface IngredientUsageModel {
    ingredientId: number;
    name: string;
    usageCount: number;
}

export interface RecipeDashboardAnalyticsModel {
    totalRecipes: number;
    recipesByStatus: { [key: string]: number };
    topRatedRecipes: RecipeSummaryModel[];
    recentlyCreated: RecipeSummaryModel[];
    mostUsedIngredients: IngredientUsageModel[];
}
